gazebo_ros/src/node.cpp

```diff
@@ -98,13 +108,36 @@ Node::SharedPtr Node::Get(sdf::ElementPtr sdf)
     }
   }
 
+  // set full node name
+  if (ns[0] == '/' && ns.size() == 1) {
+    full_name = ns + name;
+  } else {
+    full_name = ns + "/" + name;
+  }
+
+  // check if node with the same name exists already
+  if (static_existing_nodes_.check_node(full_name)) {
+    RCLCPP_ERROR(
+      internal_logger(),
+      "Found multiple nodes with same name: %s. This might be due to multiple plugins using same "
+      "name. Try changing one of the the plugin names or use a different ROS namespace. "
+      "This error might also result from a custom plugin inheriting from another gazebo_ros plugin "
+      "and the custom plugin trying to access the ROS node object hence creating multiple nodes "
+      "with same name. To solve this try providing the optional node_name argument in "
+      "gazebo_ros::Node::Get() function. ", full_name.c_str());
+    return nullptr;  // this makes the gazebo shutdown safely
```

Comment:
nitpick: technically this doesn't cause gazebo to shutdown, it is up to the plugin to check for nullptr and fail gracefully.
```suggestion
    return nullptr;
```